docs: add tql explain section (#389)
GreptimeTeam/docs: docs/reference/sql/tql.md

TQL eval (1677057993, 1677058993, '1m') rate(prometheus_http_requests_total{job="prometheus"}[5m]);
```

> Will get result just like other normal SQL queries.

Reviewer: ```suggestion
will get result just like other normal SQL queries.
```

## EXPLAIN

> `EXPLAIN` shows the logical plan and execution plan of one PromQL query. The syntax is:

Reviewer: ```suggestion
`EXPLAIN` displays both the logical plan and execution plan for a given PromQL query. The syntax is as follows:
```

```
TQL EXPLAIN expr;
```

For example, to explain the PromQL `sum by (instance) (rate(node_disk_written_bytes_total[2m])) > 50`, we can use

```
TQL EXPLAIN sum by (instance) (rate(node_disk_written_bytes_total[2m])) > 50;
```

> Notice that since the given query won't be actual executed, thus the triple `(start, end, step)` is not required. But you can still provide one like in `TQL EVAL`:

Reviewer: ```suggestion
Notice that since the given query won't be actually executed, the triple `(start, end, step)` is not necessary. But you can still provide it like in `TQL EVAL`:
```

```

## ANALYZE

> TQL also supports `ANALYZE` keyword to analyze the given PromQL query's execution. The syntax is:

Reviewer: ```suggestion
TQL also supports `ANALYZE` keyword to analyze the given PromQL query's execution. The syntax is as follows:
```

```
TQL ANALYZE (start, end, step) expr;
```

For example:

```
TQL ANALYZE (0, 10, '5s') test;
```

> Will get result like

Reviewer: ```suggestion
will get result like
```
